evm-contracts/src/v0.7/dev/OperatorForwarder.sol

```diff
@@ -4,23 +4,70 @@ pragma solidity ^0.7.0;
 import "../interfaces/OperatorInterface.sol";
 import "./ConfirmedOwner.sol";
 
-contract OperatorForwarder {
-  address public immutable authorizedSender1;
-  address public immutable authorizedSender2;
-  address public immutable authorizedSender3;
+contract OperatorForwarder is ConfirmedOwner{
+
+  mapping(address => bool) private s_authorizedSenders;
+  address[] private s_authorizedSenderList;
 
   address public immutable linkAddr;
 
+  event AuthorizedSendersChanged(
+    address[] senders
+  );
+
   constructor(
     address link
-  ) {
+  ) 
+    ConfirmedOwner(msg.sender)
+  {
     linkAddr = link;
-    authorizedSender1 = ConfirmedOwner(msg.sender).owner();
-    address[] memory authorizedSenders = OperatorInterface(msg.sender).getAuthorizedSenders();
-    authorizedSender2 = (authorizedSenders.length > 0) ? authorizedSenders[0] : address(0);
-    authorizedSender3 = (authorizedSenders.length > 1) ? authorizedSenders[1] : address(0);
   }
 
+  /**
+   * @notice Sets the fulfillment permission for a given node. Use `true` to allow, `false` to disallow.
+   * @param senders The addresses of the authorized Chainlink node
+   */
+  function setAuthorizedSenders(
+    address[] calldata senders
+  )
+    external
+    onlyOwner()
+  {
+    require(senders.length > 0, "Must have at least 1 authorized sender");
+    // Set previous authorized senders to false
+    uint256 authorizedSendersLength = s_authorizedSenderList.length;
+    for (uint256 i = 0; i < authorizedSendersLength; i++) {
+      s_authorizedSenders[s_authorizedSenderList[i]] = false;
```

Comment:
Is it not possible to delete an entire mapping in solidity?

Reply:
Unfortunately not :(